libnmstate/error.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import errno as posix_errno
+
+
+class NmstateError(Exception):
```

Comment:
Having a single error class is usually not a good thing. It require the callers to evaluate the ID/message, which is not that nice.
Having dedicated exceptions, which inherit from a base generic one will answer both challenges.

I suggest something like this:
```
class NmstateError(Exception):
    code = 0
    message = "UNKNOWN ERROR"

    def __str__(self):
        return self.message + str(self.args)

    def code(self):
        return self.code

class MyError(NmstateError):
    code = 1
    message = 'My Error Description'
```

Reply:
The reason to have single exception is when we expand error type, user application don't have to change to support this new exception.

Reply:
Yeah, use the base exception is better after I read some blog posts about this. Will try to improve the code tomorrow.